src/oneapi/oneapi.pyx

```diff
@@ -45,25 +45,30 @@ cdef class sycl_execution_context:
         del self.c_ptr
 
 
-from contextlib import contextmanager
+# thread-local storage
+from threading import local as threading_local
+_tls = threading_local()
+if getattr(_tls, 'initialized', None) is None:
```

Comment:
This is executed in the main thread, and I am not at all certain that the TLS class initialized here will end up being initialized in each additional thread.

It should be possible to find that out experimentally, but I would rather err on the side of cautious and add the logic to `def is_in_cycl_ctxt():` to check that `_tls` has `initialized` attribute, and initialize if it does not.

Reply:
done